### ISSUE TITLE
Change EE options to enum array

### ISSUE THREAD

**Issue by petertgiles** [GCTC-NTGC/gc-digital-talent#3059]
Right now the EE options need to be considered with an OR search.  This means that when building the query they have to be considered together in a subquery, not one-by-one as before.  To handle this we have factored them out into a sub-input in the graphql schema even though the actual database fields exist flatly on the candidate/user tables.  It would be more natural if these options were enums in an array, like work locations.

# Acceptance Criteria
- [ ] Migration (user and pool candidate)
  - [ ] Create new column for enums
  - [ ] Copy existing values from flat columns to enum
  - [ ]  Drop flat columns
- [ ] Update schema to use array of enums instead of nested type
- [ ] Update models to handle changed schema
- [ ] Update seeders, tests, factories
- [ ] Update frontend to handle changed schema

**Comment by vd1992** [comment]
@petertgiles 
Is this still desired, the frontend utilizing equity stuff has grown in complexity considerably since this issue was made

**Comment by petertgiles** [comment]
Yeah, the splitting out of is_indigenous kind of ruined this idea.
